packages/@uppy/audio/src/index.js

```diff
@@ -1,364 +1 @@
-    // attribute in order to determine the correct MIME type.
-    const mimeType = this.#recordingChunks.find(blob => blob.type?.length > 0).type
-
-    const fileExtension = getFileTypeExtension(mimeType)
-
-    if (!fileExtension) {
-      return Promise.reject(new Error(`Could not retrieve recording: Unsupported media type "${mimeType}"`))
-    }
-
-    const name = `audio-${Date.now()}.${fileExtension}`
-    const blob = new Blob(this.#recordingChunks, { type: mimeType })
-    const file = {
-      source: this.id,
-      name,
-      data: new Blob([blob], { type: mimeType }),
-      type: mimeType,
-    }
-
-    return Promise.resolve(file)
-  }
-
-  #changeSource = (deviceId) => {
-    this.#stop()
-    this.#start({ deviceId })
-  }
-
-  #updateSources = () => {
-    this.#mediaDevices.enumerateDevices().then(devices => {
-      this.setPluginState({
-        audioSources: devices.filter((device) => device.kind === 'audioinput'),
-      })
-    })
-  }
-
-  render () {
-    if (!this.#audioActive) {
-      this.#start()
-    }
-
-    const audioState = this.getPluginState()
-
-    if (!audioState.audioReady || !audioState.hasAudio) {
-      return (
-        <PermissionsScreen
-          icon={this.icon}
-          i18n={this.i18n}
-          hasAudio={audioState.hasAudio}
-        />
-      )
-    }
-
-    return (
-      <RecordingScreen
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        {...audioState}
-        audioActive={this.#audioActive}
-        onChangeSource={this.#changeSource}
-        onStartRecording={this.#startRecording}
-        onStopRecording={this.#stopRecording}
-        onDiscardRecordedAudio={this.#discardRecordedAudio}
-        onSubmit={this.#submit}
-        onStop={this.#stop}
-        i18n={this.i18n}
-        showAudioSourceDropdown={this.opts.showAudioSourceDropdown}
-        supportsRecording={supportsMediaRecorder()}
-        recording={audioState.isRecording}
-        stream={this.#stream}
-      />
-    )
-  }
-
-  install () {
-    this.setPluginState({
-      audioReady: false,
-      recordingLengthSeconds: 0,
-    })
-
-    const { target } = this.opts
-    if (target) {
-      this.mount(target, this)
-    }
-
-    if (this.#mediaDevices) {
-      this.#updateSources()
-
-      this.#mediaDevices.ondevicechange = () => {
-        this.#updateSources()
-
-        if (this.#stream) {
-          let restartStream = true
-
-          const { audioSources, currentDeviceId } = this.getPluginState()
-
-          audioSources.forEach((audioSource) => {
-            if (currentDeviceId === audioSource.deviceId) {
-              restartStream = false
-            }
-          })
-
-          if (restartStream) {
-            this.#stop()
-            this.#start()
-          }
-        }
-      }
-    }
-  }
-
-  uninstall () {
-    if (this.#stream) {
-      this.#stop()
-    }
-
-    this.unmount()
-  }
-}
+export { default } from './Audio.jsx'
```

Comment:
Why can't we have this exported file directly in here?

Reply:
Because the package entry point has to be `lib/index.js` (changing that would be a breaking change), and we cannot put JSX in a JS file.